src/foam/classloader/WebModelFileDAO.js

```diff
@@ -52,10 +52,19 @@ foam.CLASS({
         var model;
         var foamCLASS = foam.CLASS;
 
-        foam.CLASS = function(m) {
-          var cls = m.class ? foam.lookup(m.class) : foam.core.Model;
-          model = cls.create(m, self);
-          foam.CLASS = foamCLASS;
+        foam.CLASS = function(clsDef) {
+          var classOfModel = clsDef.class ? foam.lookup(clsDef.class) :
```

Comment:
classOfModel followed by modelOfClass is confusing, because the modelOfClass is not the model of the class referenced in classOfModel.

The confusing bit here (IMO) is that we're overriding foam.CLASS() to extract the model definition from a file which contains a foam.CLASS call.   These descriptive names do not adequately describe the "backflips" we're doing, they just make it longer which makes it even harder to parse IMO.   I would rather keep the code as is and add a small comment explaining what we're doing.

Once you understand what this code is trying to achieve I think it reads fine as is.